src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,208 @@
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(T value) {
+            this.value = value;
+        }
+
+        @Override
+        public String toString() {
+            return String.valueOf(value);
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<T>(value);
+        if (first == null) {
+            newNode.next = null;
+            newNode.prev = null;
+            first = newNode;
+            last = newNode;
+        } else {
+            last.next = newNode;
+            newNode.prev = last;
+            last = newNode;
+        }
+        listCount++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > listCount) {
+            throw new IndexOutOfBoundsException("index is wrong");
+        }
+        Node<T> newNode = new Node(value);
+        if (index == 0 && listCount == 0) {
+            add(value);
+        } else if (index == 0 && listCount > 0) {
+            newNode.next = first;
+            newNode.prev = null;
+            first.prev = newNode;
+            first = newNode;
+            listCount++;
+        } else if (index == listCount) {
+            add(value);
+        } else {
+            Node<T> oldNode = first;
+            for (int i = 0; i < index; i++) {
+                oldNode = oldNode.next;
+            }
+            addItem(oldNode, newNode);
+            listCount++;
+
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= listCount) {
+            throw new IndexOutOfBoundsException("index is wrong");
+        }
+        Node<T> node = first;
+        if (index == 0) {
+            return (T) first.value;
+        }
+        for (int i = 0; i < index; i++) {
+            node = node.next;
+        }
+        return (T) node.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (index < 0 || index >= listCount) {
+            throw new IndexOutOfBoundsException("index is wrong");
+        }
+        Node<T> oldNode = first;
+        Node<T> newNode = new Node<>(value);
+        if (index == 0) {
+            newNode.next = first.next;
+            newNode.prev = first.prev;
+            first.next.prev = newNode;
+            first = newNode;
+        } else if (index == listCount - 1) {
+            oldNode = last;
+            newNode.next = null;
+            newNode.prev = last.prev;
+            last.prev.next = newNode;
+            newNode = last;
+        } else {
+            for (int i = 0; i < index; i++) {
+                oldNode = oldNode.next;
+            }
+            addItem(oldNode, newNode);
+        }
+        return (T) oldNode.value;
```

Comment:
too complex logic, this is how this method logic should look like:
1. Find node by index.
2. Create a variable that will store the current value of found node.
3. Update the value of the current node with a new one.
4. Return old value of a node